### PR TITLE
[Motions 2023 11 cwg 4] P2864R2 Remove deprecated arithmetic conversions

### LINKED ISSUE #6658: [2023-11 CWG Motion 4] Remove Deprecated Arithmetic Conversion on Enumerations From C++26

### PR DESCRIPTION
**Pull request by AlisdairM**
Fixes #6658 
Fixes cplusplus/papers#1554

### CODE REVIEW

File: source/compatibility.tex [hunk 1]
```diff
@@ -2463,6 +2481,31 @@
...
+int b = e <= 3.7;       // valid in C; ill-formed in \Cpp{}
+int k = f - e;          // valid in C; ill-formed in \Cpp{}
+int x = 1 ? e : f;      // valid in C; ill-formed in \Cpp{}
+\end{codeblock}
+\rationale
+Reinforcing type safety in \Cpp{}.
+\effect
+Well-formed C code will not compile with this International Standard.
+\difficulty
+\begin{codeblock}
```
Comment (JohelEGP) [inline source/compatibility.tex:2500]: There's missing text under `\difficulty`.

**Comment by burblebee** [comment]
@AlisdairM, your changes look good other than:
* Your wording changes to the `\effect` clause diverge a bit much from the paper. At the very least, this divergence should have been included with the initial commit comment and explained, but preferably, a change this great from the original wording should have been done in a separate commit.
* As Johel mentioned, you're missing the wording for `\difficulty`.
* The commit comments for changes which should have been included in the original commit (i.e. your fixes to what Johel found) should be prefixed with "FIXUP" so that the editor knows to squash these before merging. e.g. your "Update source/compatibility.tex" comments should have been something like "FIXUP: Add missing punctuation after "Uncommon".

File: source/compatibility.tex (hunk 1 above)
Comment (tkoeppe) [reply source/compatibility.tex:2559]: Thanks, fixed!

File: source/compatibility.tex [hunk 2]
```diff
@@ -2522,6 +2541,35 @@
...
+int b = e <= 3.7;       // valid in C; ill-formed in \Cpp{}
+int k = f - e;          // valid in C; ill-formed in \Cpp{}
+int x = 1 ? e : f;      // valid in C; ill-formed in \Cpp{}
+\end{codeblock}
+\rationale
+Reinforcing type safety in \Cpp{}.
+\effect
+Well-formed C code will not compile with this International Standard.
+\difficulty
+Violations will be diagnosed by the C++ translator.
```
Comment (JohelEGP) [inline source/compatibility.tex:2561]:
```suggestion
Violations will be diagnosed by the \Cpp{} translator.
```

**Pull request merged**